db/cache/charts.go

```diff
@@ -1164,3 +1184,12 @@ func poolValueChart(charts *ChartData, bin binLevel, axis axisType) ([]byte, err
 	}
 	return nil, InvalidBinErr
 }
+
+func missedVotesChart(charts *ChartData, _ binLevel, _ axisType) ([]byte, error) {
+	stakeValidWindows := int(charts.StartPOS / charts.DiffInterval)
+	if stakeValidWindows >= len(charts.Windows.MissedVotes) {
+		stakeValidWindows = 0
+	}
+	return charts.encode(charts.Windows.Time[stakeValidWindows:],
```

Comment:
Please add a check that `stakeValidWindows` is in range.

Reply:
I thought this should work:
```golang
        if stakeValidWindows >= len(charts.Windows.MissedVotes) {
		stakeValidWindows = 0
	}
```

Reply:
Adding `charts.Windows.Time` too.

Reply:
This one should work:
```golang
	if stakeValidWindows >= len(charts.Windows.MissedVotes) ||
		stakeValidWindows >= len(charts.Windows.Time) {
		stakeValidWindows = 0
	}
```